src/routes/getFiles.ts

```diff
@@ -0,0 +1,235 @@
+  ignoreTimestamps: asOptional(asBoolean),
+  paths: asMap(asNumber)
+})
+
+interface GetFilesResponseData {
+  total: number
+  paths: GetFilesResponsePaths
+}
+
+interface GetFilesResponsePaths {
+  [path: string]: GetFilesResponseFile | GetFilesResponseDirectory
+}
+
+interface GetFilesResponseFile extends StoreFile {
+  timestamp: number
+}
+
+interface GetFilesResponseDirectory {
+  files: {
+    [path: string]: number
+  }
+  timestamp: number
+}
+
+export const getFilesRouter = Router()
+
+getFilesRouter.post('/getFiles', async (req, res) => {
+  let body: GetFilesBody
+  let repoId: string
+  let ignoreTimestamps: boolean = false
+
+  // Validate request body
+  try {
+    body = asGetFilesBody(req.body)
+    repoId = body.repoId
+
+    if (body.ignoreTimestamps !== undefined) {
+      ignoreTimestamps = body.ignoreTimestamps
+    }
+
+    if (repoId === '') {
+      throw new Error(`Missing repoId.`)
+    }
+  } catch (err) {
+    throw makeApiClientError(400, err.message)
+  }
+
+  const responsePaths: GetFilesResponsePaths = {}
+
+  const paths = Object.keys(body.paths)
+
+  // Use max page size config to limit the paths processed
+  if (paths.length > MAX_PAGE_SIZE) {
+    throw makeApiClientError(
+      422,
+      `Too many paths. MAX_PAGE_SIZE is ${MAX_PAGE_SIZE}`
+    )
+  }
+
+  const keys = paths.map(filePath => `${repoId}:${filePath}`)
+  // Maps a directory key to a file path for timestamp lookup
+  const filePathMapOfParentDirKey: { [dirKey: string]: string } = {}
+
+  // Fetch Paths:
+
+  const fetchPathsResults = await dataStore.fetch({ keys })
+
+  for (const row of fetchPathsResults.rows) {
+    const documentKey = row.key
+    // No need to check for undefined case here because of how keys
+    // are defined for the fetch.
+    const documentPath = documentKey.split(':')[1]
+
+    const requestTimestamp = body.paths[documentPath]
+
+    if ('doc' in row) {
+      let fileDocument: StoreFileDocument | undefined
+      let directoryDocument: StoreDirectoryDocument | undefined
+
+      try {
+        // For file document
+        fileDocument = asStoreFileDocument(row.doc)
+      } catch (err) {
+        try {
+          // For directory document
+          directoryDocument = asStoreDirectoryDocument(row.doc)
+        } catch (err) {
+          throw makeApiClientError(
+            422,
+            `File '${row.key}' is not a file or directory.`
+          )
+        }
+      }
+
+      // Handle file document
+      if (fileDocument !== undefined) {
+        responsePaths[documentPath] = {
+          ...asStoreFile(fileDocument),
+          timestamp: 0
+        }
+
+        // Add file path to file directory map for timestamp lookup
+        if (!ignoreTimestamps) {
+          const [parentDirectoryPath = '/'] = getParentPathsOfPath(documentPath)
+          const parentDirectoryKey = `${repoId}:${parentDirectoryPath}`
+          filePathMapOfParentDirKey[parentDirectoryKey] = documentPath
+        }
```

Comment:
1. This should not check for ignoreTimestamps here because we will always need to get the file's timestamp from it's parent.
2. We must set `filePathMapOfParentDirKey` element to an new array with `documentPath` element or push it into the existing array here.